### PR TITLE
[TIMOB-24680] (6_1_X) Fixed list of valid provisioning profiles for the given…

### PR DESCRIPTION
**Pull request by cb1kenobi**
[TIMOB-24680] Fixed list of valid provisioning profiles for the given build target which is used to validate watch extension provisioning profile uuids.

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24680

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -1434,6 +1434,26 @@ iOSBuilder.prototype.initTiappSettings = function initTiappSettings() {
...
+
+		function getPPbyUUID(ppuuid) {
+			return pps
+				.filter(function (p) {
+					if (!p.expired && !p.managed && p.uuid === ppuuid) {
+						return true;
+					}
+				})
+				.shift();
+		}
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: We use that method too often. Please add this helper to our utils instead of pasting it to every usage.

**Review by hansemannn — CHANGES_REQUESTED** [review]

**Review by hansemannn — APPROVED** [review]

File: iphone/cli/commands/_build.js (hunk 1 above)
Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: Great idea for the next overhaul.

**Review by htbryant94 — APPROVED** [review]
Approved! Enterprise Adhoc Provisioning Profiles are now validated correctly for Watch Extensions, and build failures no longer occur.

**Comment by hansemannn** [comment]
@htbryant94 Wait a second, should this go into 6.1.1? Because you just merged into 6_1_X, not master. The ticket says 6.2.0, however, I'd be fine for a 6.1.1 if @eric34 approves.

**Comment by htbryant94** [comment]
@hansemannn I asked @eric34 whether to merge the fix into the 6_1_X branch as well as the master, and was confirmed to do so. It would be good to get a second approval just in case, however I do not see any reason why it shouldn't go into 6.1.1

**Pull request merged**